packages/ui/src/App.tsx

```diff
@@ -6,22 +6,53 @@ import ReactMarkdown from 'react-markdown';
 import {Prism as SyntaxHighlighter} from 'react-syntax-highlighter';
 import axios from 'axios';
 import {ExpansionPanel, ExpansionPanelDetails, ExpansionPanelSummary, Link, makeStyles, Typography} from '@material-ui/core';
+import {useDebounce} from 'use-debounce';
+
+type SetQuery = (x: DebouncableQuery) => void;
+const getQueryFromSearch = (search: string): string =>
+    decodeURIComponent(
+        search
+            .slice(1)
+            .split('&')
+            .find((param) => param.startsWith('q='))
+            ?.split('=')[1] ?? ''
+    );
+
+interface DebouncableQuery {
+    query: string;
+    debounce: boolean;
+}
+
+const useUrlChangableQuery = (): [DebouncableQuery, (v: DebouncableQuery) => void] => {
```

Comment:
```suggestion
const useUrlChangeableQuery = (): [DebouncableQuery, (v: DebouncableQuery) => void] => {
```
Bin nicht sicher, ob man nicht auch eher `Debaunceable` schreiben würde. Aber `Changeable` ist zumindest ein existierendes Wort.